app/graphql/mutations/create_sample.rb

```diff
@@ -15,37 +15,63 @@ class CreateSample < BaseMutation
              description: 'Persistent Unique Identifier of the project. For example, `INXT_PRJ_AAAAAAAAAA`.'
     validates required: { one_of: %i[project_id project_puid] }
 
-    field :errors, [String], null: false, description: 'A list of errors that prevented the mutation.'
+    field :errors, [Types::UserErrorType], null: false, description: 'A list of errors that prevented the mutation.'
     field :sample, Types::SampleType, description: 'The newly created sample.'
 
-    def resolve(args) # rubocop:disable Metrics/MethodLength
-      project = if args[:project_id]
-                  IridaSchema.object_from_id(args[:project_id], { expected_type: Project })
-                else
-                  Namespaces::ProjectNamespace.find_by!(puid: args[:project_puid]).project
-                end
-      sample = Samples::CreateService.new(current_user, project,
-                                          { name: args[:name], description: args[:description] }).execute
+    def resolve(args)
+      project = get_project(args)
+
+      if project.nil? || !project.persisted?
+        user_errors = [{
+          path: ['project'],
+          message: 'Project not found by provided ID or PUID'
+        }]
+        return {
+          sample: nil,
+          errors: user_errors
+        }
+      end
+
+      create_sample(project, args)
+    end
+
+    def ready?(**_args)
+      authorize!(to: :mutate?, with: GraphqlPolicy, context: { user: context[:current_user], token: context[:token] })
+    end
+
+    private
+
+    def get_project(args)
+      if args[:project_id]
```

Comment:
**nitpick (blocking):** Same comment as in `app/graphql/mutations/attach_files_to_sample.rb`

Reply:
Changed in https://github.com/phac-nml/irida-next/pull/669/commits/f2a3ab4afd9aefb41993583921f38d5c6095ab29 and https://github.com/phac-nml/irida-next/pull/669/commits/c06ff5bccf93c44e02a02ad3006389726e563f3e